mne/io/cnt/cnt.py

```diff
@@ -272,18 +352,25 @@ def _get_cnt_info(input_fname, eog, ecg, emg, misc, data_format, date_format):
     for ch, loc in zip(chs, locs):
         ch.update(loc=loc)
 
-    # Add the stim channel.
-    chan_info = {'cal': 1.0, 'logno': len(chs) + 1, 'scanno': len(chs) + 1,
-                 'range': 1.0, 'unit_mul': 0., 'ch_name': 'STI 014',
-                 'unit': FIFF.FIFF_UNIT_NONE,
-                 'coord_frame': FIFF.FIFFV_COORD_UNKNOWN, 'loc': np.zeros(12),
-                 'coil_type': FIFF.FIFFV_COIL_NONE, 'kind': FIFF.FIFFV_STIM_CH}
-    chs.append(chan_info)
-    baselines.append(0)  # For stim channel
-    cnt_info.update(baselines=np.array(baselines), n_samples=n_samples,
-                    stim_channel=stim_channel, n_bytes=n_bytes)
+    if stim_channel_toggle:
+        chan_info = {'cal': 1.0, 'logno': len(chs) + 1, 'scanno': len(chs) + 1,
+                     'range': 1.0, 'unit_mul': 0., 'ch_name': 'STI 014',
+                     'unit': FIFF.FIFF_UNIT_NONE,
+                     'coord_frame': FIFF.FIFFV_COORD_UNKNOWN,
+                     'loc': np.zeros(12),
+                     'coil_type': FIFF.FIFFV_COIL_NONE,
+                     'kind': FIFF.FIFFV_STIM_CH}
+        chs.append(chan_info)
+        baselines.append(0)  # For stim channel
+        cnt_info.update(baselines=np.array(baselines), n_samples=n_samples,
+                        stim_channel=stim_channel, n_bytes=n_bytes)
+    else:
+        cnt_info.update(baselines=np.array(baselines), n_samples=n_samples,
+                        n_bytes=n_bytes)
```

Comment:
There is redundancy in several conditionals, including here, which could instead be (no need for `else` at all):
```
if stim_channel_toggle:
    ...
    cnt_info.update(stim_channel=stim_channel)
cnt_info.update(baselines=np.array(baselines), n_samples=n_samples,
                n_bytes=n_bytes)
```
Seems simpler (makes it clear no matter what, `cnt_info` is updated with these entries) and is more DRY, no?

Reply:
totally.